src/utils/signaling.js

```diff
@@ -496,6 +516,23 @@ Signaling.Internal.prototype.sendPendingMessages = function() {
 	}.bind(this))
 }
 
+Signaling.Internal.prototype._leaveRoomBefore = function(token) {
+	console.log('Signaling.Internal._leaveRoomBefore() token=' + token +
+		' currentRoomToken=' + this.currentRoomToken + ' pullMessagesRequest.token=' + this.pullMessagesRequest?.token)
+	// note: we can't rely on this.currentRoomToken because in some cases it's already cleared
+	if (this.pullMessagesRequest?.token === token) {
+	//if (this.pullMessagesRequest) {
```

Comment:
using this one seems to make the 404 errors disappear completely in my tests

but there's a risk that we might cancel too eagerly and find ourselves in a situation where no more messages are getting pulled